lib/pbench/server/api/resources/upload_api.py

```diff
@@ -190,3 +194,25 @@ def allowed_file(filename):
             return False
         allowed = "." in filename and fn in ALLOWED_EXTENSIONS
         return allowed
+
+    @property
+    def upload_directory(self):
+        prdp = self.config.get_conf(
+            __name__, "pbench-server", "pbench-receive-dir-prefix", self.logger
+        )
+        try:
+            return Path(f"{prdp}-002").resolve(strict=True)
```

Comment:
Since we are returning a `Path` object here, line 106 can be changed to `path = self.upload_directory / controller`.